### ISSUE TITLE
[Feature Request] Add workaround to work with gcc 12

### ISSUE THREAD

**Issue by c0h2r** [vial-kb/vial-qmk#189]
## Feature Request Type

- [x] Core functionality
- [ ] Add-on hardware support (eg. audio, RGB, OLED screen, etc.)
- [ ] Alteration (enhancement/optimization) of existing feature(s)
- [ ] New behavior

## Description
On arch (and maybe some other rolling release distros) the current version of gcc is 12.1, and there's a bug ( https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105523 ) which results in a bunch of `error: array subscript 0 is outside array bounds of ...` here and there.  The problem is solved in the main qmk_firmware repo with 3 new lines of code in `platforms/avr/platform.mk`.

Basically, it'd be geat if you reflect the changes made with the following pull request:
- https://github.com/qmk/qmk_firmware/pull/17136

And for now those who have the same problem can just go to the files section and copy-paste those 3 lines in the corresponding file.

**Comment by xyzz** [comment]
Should be resolved now
